### PR TITLE
Publish to Bintray

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
@markschaake I would appreciate a look-over from you too.  I'm pretty convinced this will work for common.  I'm a little worried about what defaults I'm causing other builds to have.

**Comment by schmmd** [comment]
Should I remove `bintray.Plugin.bintrayPublishSettings` so downstream applications specify this instead?  Is it OK to leave the bintray plugin in the allenai plugins?

**Comment by markschaake** [comment]
Sorry, just noticed this... taking a look now

File: src/main/scala/org/allenai/plugins/ReleasePlugin.scala [hunk 1]
```diff
@@ -62,11 +63,43 @@ object ReleasePlugin extends AutoPlugin {
...
-      releaseVersion := { version => SemanticVersion.computeReleaseVersion(version) },
-      nextVersion := { version => SemanticVersion.computeNextVersion(version) }
+      ReleaseKeys.releaseVersion := { version => SemanticVersion.computeReleaseVersion(version) },
+      ReleaseKeys.nextVersion := { version => SemanticVersion.computeNextVersion(version) }
     )
   }
 
-  override lazy val projectSettings: Seq[Def.Setting[_]] =
-    WrappedReleasePlugin.releaseSettings ++ DateVersion.settings
+  val checkBranchIsNotMaster = { st: State =>
```
Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/ReleasePlugin.scala]: A type on this `(State => State)` would be really helpful.

File: src/main/scala/org/allenai/plugins/ReleasePlugin.scala [hunk 2]
```diff
@@ -62,11 +63,43 @@ object ReleasePlugin extends AutoPlugin {
...
+    val vcs = Project.extract(st).get(ReleaseKeys.versionControlSystem).getOrElse {
+      sys.error("Aborting release. Working directory is not a repository of a recognized VCS.")
+    }
+
+    if (vcs.currentBranch == "master") {
+      sys.error("Current branch is master.  At AI2, releases are done from another branch and " +
```
Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/ReleasePlugin.scala]: If you were writing this to be more open-source / user friendly, you could make the deploy branch be a setting key.

**Comment by jkinkead** [comment]
A couple of small comments. I'm not sure what adding the bintray settings you're adding do, so I don't know if they'd be harmful at all.

File: src/main/scala/org/allenai/plugins/ReleasePlugin.scala (hunk 2 above)
Comment (schmmd) [inline src/main/scala/org/allenai/plugins/ReleasePlugin.scala]: This specifies which branch CANNOT be used for releases.  Setting is a good idea--I'll see how easy that is for me.

**Comment by markschaake** [comment]
I also don't know what adding the bintray settings will do. We could just see what happens :)

LGTM

**Pull request merged**